Localize runtime #2335 (#2374)
cncf/glossary: content/it/runtime.md

---
title: Runtime
status: Completed
category: concetto
tags: ["applicazione", "", ""]
---

## Cos'è

Un runtime, in generale, esegue un software.
È un'[astrazione](/it/abstraction/) del sistema operativo sottostante che traduce i comandi del programma in azioni relative per il sistema operativo.

Nel contesto [cloud native](/it/cloud-native-apps/), per _runtime_ si intende container runtime. 
Un container runtime è un implementazione che segue le specifiche [Open Container Initiative](https://opencontainers.org/) per garantire una gestione coerente dei container nelle differenti tecnologie di orchestrazione degli stessi.

## Quali problematiche affronta
Senza l'astrazione di un runtime del container, l'applicazione dovrebbe occuparsi di tutte le meccaniche di ciascun sistema operativo, aumentando la complessità dell'esecuzione dell'app.

## In che modo aiuta

> I container runtime sono un componente necessario degli orchestratori di container come Kubernetes. 

Reviewer: ```suggestion
I container runtime sono un componente necessario degli orchestratori di container come [Kubernetes](/it/kubernetes). 
```

Reviewer: Non c'è nell'originale ma secondo me ci sta bene

Gestiscono il ciclo di vita del container, che fa principalmente tre cose: 

> definisce come vengono specificate le immagini del container e come il runtime può recuperarle;
> gestiscono come queste immagini vengono decompresse, stratificate, montate ed eseguite;
> gestiscono le risorse hardware occupandosi di tutte queste azioni a livello di sistema operativo, come allocazione e isolamento delle risorse.

Reviewer: ```suggestion

- definisce come vengono specificate le immagini del container e come il runtime può recuperarle;
- gestisce come queste immagini vengono decompresse, stratificate, montate ed eseguite;
- gestisce le risorse hardware occupandosi di tutte queste azioni a livello di sistema operativo, come allocazione e isolamento delle risorse.

```

> Nel tempo, diversi implementazioni di container runtime hanno continuato ad evolversi, portando alla Specifica OCI, che è diventata lo standard per i runtime dei container.

Reviewer: ```suggestion
Nel tempo, diverse implementazioni di container runtime hanno continuato ad evolversi, portando alla Specifica OCI, che è diventata lo standard per i runtime dei container.
```

---
title: Runtime
status: Completed
category: concetto
tags: ["applicazione", "", ""]
---

## Cos'è

Un runtime, in generale, esegue un software.

> È un'[astrazione](/it/abstraction/) del sistema operativo sottostante che traduce i comandi del programma in azioni relative per il sistema operativo.

Reviewer 2: ```suggestion
È un'[astrazione](/it/abstraction/) del sistema operativo sottostante che traduce i comandi del programma in azioni corrispondenti perché siano utili al sistema operativo stesso.
```

> Nel contesto [cloud native](/it/cloud-native-apps/), per _runtime_ si intende container runtime. 

Reviewer 2: ```suggestion
Nel contesto [cloud native](/it/cloud-native-apps/), il termine runtime si riferisce generalmente al runtime dei container. 
```

Un container runtime è un implementazione che segue le specifiche [Open Container Initiative](https://opencontainers.org/) per garantire una gestione coerente dei container nelle differenti tecnologie di orchestrazione degli stessi.

## Quali problematiche affronta

> Senza l'astrazione di un runtime del container, l'applicazione dovrebbe occuparsi di tutte le meccaniche di ciascun sistema operativo, aumentando la complessità dell'esecuzione dell'app.

Reviewer 2: ```suggestion
Senza l'astrazione di un runtime del container, l'applicazione dovrebbe occuparsi di tutte le meccaniche di ciascun sistema operativo, aumentando la complessità dell'esecuzione dell'applicazione.
```
